Mage.Sets/src/mage/cards/r/RoryWilliams.java

```diff
@@ -0,0 +1,101 @@
+import mage.abilities.keyword.*;
+import mage.constants.*;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.stack.Spell;
+
+/**
+ *
+ * @author Skiwkr
+ */
+public final class RoryWilliams extends CardImpl {
+
+    public RoryWilliams(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{W}{U}");
+        
+        this.supertype.add(SuperType.LEGENDARY);
+        this.subtype.add(SubType.HUMAN);
+        this.subtype.add(SubType.SOLDIER);
+        this.power = new MageInt(3);
+        this.toughness = new MageInt(3);
+
+        // Partner with Amy Pond
+        this.addAbility(new PartnerWithAbility("Amy Pond"));
+
+        // First strike
+        this.addAbility(FirstStrikeAbility.getInstance());
+
+        // Lifelink
+        this.addAbility(LifelinkAbility.getInstance());
+
+        // The Last Centurion -- When you cast this spell from anywhere other than exile, exile it with three time counters on it. It gains suspend. Then investigate.
+        Ability ability = new RoryWilliamsTriggeredAbility(new ExileSpellWithTimeCountersEffect(3, true));
+        ability.addEffect(new InvestigateEffect(1).concatBy("Then"));
+        ability.withFlavorWord("The Last Centurion");
+        this.addAbility(ability);
+    }
+
+    private RoryWilliams(final RoryWilliams card) {
+        super(card);
+    }
+
+    @Override
+    public RoryWilliams copy() {
+        return new RoryWilliams(this);
+    }
+}
+
+class RoryWilliamsTriggeredAbility extends CastSourceTriggeredAbility {
+
+    protected RoryWilliamsTriggeredAbility(Effect effect) {
+        super(effect, false);
+        this.ruleAtTheTop = true;
+        setTriggerPhrase("When you cast this spell from anywhere other than exile, ");
+    }
+
+    protected RoryWilliamsTriggeredAbility(final RoryWilliamsTriggeredAbility ability) {
+        super(ability);
+    }
+
+    @Override
+    public RoryWilliamsTriggeredAbility copy() {
+        return new RoryWilliamsTriggeredAbility(this);
+    }
+
+    @Override
+    public boolean checkEventType(GameEvent event, Game game) {
+        return event.getType() == GameEvent.EventType.SPELL_CAST;
+    }
+
+    @Override
+    public boolean checkTrigger(GameEvent event, Game game) {
+        if (!event.getSourceId().equals(this.getSourceId())) {
+            return false;
+        }
+        MageObject spellObject = game.getObject(sourceId);
+        if ((!(spellObject instanceof Spell))) {
+            return true;
+        }
+        Spell spell = (Spell) spellObject;
+        if (spell.getSpellAbility() != null) {
+            getEffects().setValue(SOURCE_CAST_SPELL_ABILITY, spell.getSpellAbility());
+        }
+        if (spell.getFromZone().equals(Zone.EXILED)){
+            return false;
+        }
+        getEffects().setValue("spellCast", spell);
+        return true;
```

Comment:
again, don't duplicate code. you're extending a class, call the super and only add what you need